lib/common/client/middleware/request/rescue_timeout.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Middleware
+      module Request
+        # must be added right before http_client adapter
+        class RescueTimeout < Faraday::Middleware
+          include SentryLogging
+
+          def initialize(app = nil, error_code = 'VA900')
```

Comment:
Allow error_code parameter so that this middleware can be plugged in to various services